DependencyInjection/Compiler/CommandHandlerPass.php

```diff
@@ -19,25 +21,100 @@ class CommandHandlerPass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
-        if (!$container->has('tactician.handler.locator.symfony')) {
-            throw new \Exception('Missing tactician.handler.locator.symfony definition');
-        }
-
-        $handlerLocator = $container->findDefinition('tactician.handler.locator.symfony');
-
-        $mapping = [];
+        $defaultBusId = $this->getDefaultBusId($container);
+        $busIdToHandlerMapping = [];
 
         foreach ($container->findTaggedServiceIds('tactician.handler') as $id => $tags) {
+
             foreach ($tags as $attributes) {
                 if (!isset($attributes['command'])) {
                     throw new \Exception('The tactician.handler tag must always have a command attribute');
                 }
 
-                $mapping[$attributes['command']] = $id;
+                $busId = array_key_exists('bus', $attributes) ? $attributes['bus'] : $defaultBusId;
+
+                $this->abortIfInvalidBusId($busId, $container);
+
+                $busIdToHandlerMapping[$busId][$attributes['command']] = $id;
```

Comment:
I'm wondering a bit about the BC scenario here. In the past, the way the bundle worked is that each handler was attached to each bus (if you had multiple). Now they'll be attached only to the default, unless you add the "bus" attribute, which is going to make for a pretty surprising BC break.

What would you two think about changing this slightly so that rather than defaulting to bus id when there's no bus attribute, that we instead attach it to all buses until otherwise specified? If you're sick of changing this one, I can make a separate PR for it. :smile: